openfeature/client_test.go

```diff
@@ -27,13 +27,35 @@ func TestRequirement_1_2_1(t *testing.T) {
 	}
 }
 
+func newTestClientWithProvider(clientName string, t *testing.T) (*Client, *MockFeatureProvider, *MockevaluationImpl) {
+	client, evalApi := newTestClient(clientName, t)
+
+	mockProvider := NewMockFeatureProvider(evalApi.ctrl)
+	mockProvider.EXPECT().Metadata().AnyTimes()
+	mockProvider.EXPECT().Hooks().AnyTimes()
+
+	evalApi.EXPECT().ForEvaluation(client.metadata.name).Times(1).DoAndReturn(func(_ string) (*MockFeatureProvider, []Hook, EvaluationContext) {
```

Comment:
Now we're even checking if this call only happens once - i am not sure if i could return without a function, at least i did not make it work